backend/python/tapa/verilog/xilinx/module.py

```diff
@@ -578,6 +587,14 @@ def cleanup(self) -> None:
     self.add_pipeline(self.rst_n_q, init=RST_N)
     self.add_logics([ast.Assign(left=RST, right=ast.Unot(self.rst_n_q[-1]))])
 
+  def _get_nodes_of_type(self, node, *target_types) -> Iterator:
+    if isinstance(node, target_types):
+      yield node
+    for c in node.children():
+      yield from self._get_nodes_of_type(c, *target_types)
+
+  def get_nodes_of_type(self, *target_types) -> Iterator:
+    yield from self._get_nodes_of_type(self.ast, *target_types)
```

Comment:
HLS-generated RTL is rather regular. I don't think we need the recursion.

Reply:
This is for the traverse of all nodes of the given type in the ast

Reply:
Yes, my point is this traversal should not be necessary.